Translate docs/contributing/how-to-open-a-pull-request.md to pt-br (#29)
gatsbyjs/gatsby-pt-BR: docs/contributing/how-to-open-a-pull-request.md

---

> title: Como Abrir uma Pull Request

Reviewer: Acho que é mais comum o artigo "um Pull Request" ao invés "uma Pull Request". Eu pelo menos conversando em conversas de IT sempre falei "um".

Author: Hey @Dennissiq, eu concordo com você, também utilizo mais "um Pull Request" ao invés "uma Pull Request". Porém eu tomei essa decisão pois encontrei escrito dessa maneira em outros lugares aqui na doc: 

> abri uma Pull Request 
> revise a PR
> como abrir uma PR
https://github.com/gatsbyjs/gatsby-pt-BR/blob/master/CONTRIBUTING.md#finalizei-a-tradu%C3%A7%C3%A3o

Além de encontrar da mesma maneira nas docs do GitHub:

> As pull requests
https://help.github.com/pt/github/collaborating-with-issues-and-pull-requests/about-pull-requests

Perante isso, devo realmente alterar de "a/uma PR" para "o/um PR"?

Author: Done: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

Reviewer: ah ok, pode deixar assim então ;) 
Acho que os dois podem ser válidos, como se trata de uma palavra em inglês acho que blz

---

> Uma boa parte de contribuição para open source consiste em enviar alterações para um projeto: melhorias no código-fonte ou nos testes, atualizações no conteúdo da documentação, ou até mesmo erros de digitação ou links quebrados. Este documento abordará o que você precisa saber para **abrir uma pull request** no Gatsby.

Reviewer: colocar em itálico open source => _open source_ 
Alterar artigo "uma" => "um"

```suggestion
Uma boa parte de contribuição para _open source_ consiste em enviar alterações para um projeto: melhorias no código-fonte ou nos testes, atualizações no conteúdo da documentação, ou até mesmo erros de digitação ou links quebrados. Este documento abordará o que você precisa saber para **abrir um pull request** no Gatsby.
```

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/f5af213ee6289a6e0645e9f7d7ef16f0b3cfe22d
https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

> ## O que é uma Pull Request (PR)?

Reviewer: Alterar artigo "uma" => "um"

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

> Caso você não esteja familiarizado, veja como o pessoal do GitHub [define uma pull request](https://help.github.com/pt/articles/about-pull-requests):

Reviewer: Alterar artigo "uma" => "um"

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

> > As pull requests permitem que você informe outras pessoas sobre as alterações das quais você fez push para um branch em um repositório no GitHub. Depois que uma pull request é aberta, você pode discutir e revisar as possíveis alterações com colaboradores e adicionar commits de acompanhamento antes que as alterações sofram merge no branch base.

Reviewer: Alterar artigo "As" => "Os"
Alterar artigo "uma" => "um"
Alterar artigo "um branch" => "uma branch"

```suggestion
> Os pull requests permitem que você informe outras pessoas sobre as alterações das quais você fez push para uma branch em um repositório no GitHub. Depois que um pull request é aberto, você pode discutir e revisar as possíveis alterações com colaboradores e adicionar commits de acompanhamento antes que as alterações sofram merge na branch base.
```

Author: Essa é uma citação, eu trouxe o texto na integra:
![image](https://user-images.githubusercontent.com/5200384/68024174-1768b280-fc88-11e9-8b7f-d4e544d33e19.png)
https://help.github.com/pt/github/collaborating-with-issues-and-pull-requests/about-pull-requests

Devo alterá-lo mesmo assim?

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa
https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/b497fb691d6b7d2acfce84e6cf0851f882f05cf3

> Gatsby usa o processo de PR para revisar e testar as alterações antes de serem adicionadas ao repositório do Gatsby no GitHub. Qualquer pessoa pode abrir uma pull request. O mesmo processo é usado para todos os contribuidores, seja esta a sua primeira contribuição open source ou seja você um membro da equipe core do Gatsby.

Reviewer: Alterar artigo "uma" => "um"

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

Quando alguém desejar contribuir com o Gatsby, essa pessoa abre uma solicitação para inserir (_pull_) seu código no repositório. Dependendo do tipo de alteração, os PRs são categorizados em:

- [Documentação](#documentation)
- [Código](#code-changes)
- [Starters or Site Showcase](#starters-or-site-showcase)
- [Blog posts](#blog-posts)

Recomendações para diferentes tipos de contribuições seguirão neste guia e ao longo das documentações de contribuição.

> ## O que você deve saber antes de abrir uma PR

Reviewer: Alterar artigo "uma" => "um"

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

Normalmente recomendamos [abrir uma issue](/contributing/how-to-file-an-issue/) antes de um pull request, se ainda não houver uma issue para o problema que você deseja resolver. Isso ajuda a facilitar uma discussão antes de decidir sobre uma implementação.

Para algumas alterações, como correções de erros de digitação ou links quebrados, pode ser apropriado abrir um pequeno PR por si só. Isso é um pouco subjetivo, portanto, se você tiver alguma dúvida, [não hesite em perguntar-nos](/contributing/how-to-contribute/#not-sure-how-to-start-contributing).

> A equipe core do Gatsby usa um processo de triagem descrito em [Gerenciando Pull Requests](/contributing/managing-pull-requests/), se você estiver interessado em aprender mais sobre como isso funciona.

Reviewer 2: ```suggestion
A equipe principal do Gatsby usa um processo de triagem descrito em [Gerenciando Pull Requests](/contributing/managing-pull-requests/), caso você tenha interesse em aprender mais sobre como isso funciona.
```

## Abrindo PRs no Gatsby

> Para qualquer tipo de alteração nos arquivos no repositório do Gatsby, você pode seguir as etapas abaixo. Certifique-se de verificar as dicas adicionais para contribuir para várias partes do repo mais adiante neste documento, como mudanças nas docs, blog posts, starters, ou melhorias de código e testes.

Reviewer 2: ```suggestion
Para qualquer tipo de alteração nos arquivos do repositório do Gatsby, sugerimos que você siga as etapas abaixo. Certifique-se de verificar as dicas adicionais para contribuir para várias partes do repositório mais adiante neste documento, como mudanças nas documentações, postagens do blog, _starters_, ou melhorias de código e testes.
```

Alguns PRs podem ser feitos completamente a partir do [GitHub UI](https://help.github.com/pt/articles/creating-a-pull-request), como edições em arquivos README ou documentações.

> Para testar alterações localmente no [site e arquivos do projeto] do Gatsby, você pode fazer um fork do repositório e instalar partes dele para executar na sua máquina local.

Reviewer: Não retirar link

```suggestion
Para testar alterações localmente no [site e arquivos do projeto](https://github.com/gatsbyjs/gatsby) do Gatsby, você pode fazer um fork do repositório e instalar partes dele para executar na sua máquina local. 
```

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

> Para testar alterações localmente no [site e arquivos do projeto](https://github.com/gatsbyjs/gatsby) do Gatsby, você pode fazer um fork do repositório e instalar partes dele para executar na sua máquina local.

Reviewer 2: ```suggestion
Para testar alterações localmente no [site e em arquivos do projeto](https://github.com/gatsbyjs/gatsby) do Gatsby, você pode fazer um fork do repositório e instalar partes dele para executar na sua máquina local.
```

- [Crie um fork e clone o repositório Gatsby](/contributing/setting-up-your-local-dev-environment/#gatsby-repo-install-instructions).

> - Instale o [yarn](https://yarnpkg.com/pt-BR/) para obter dependências e criar o projeto.

Reviewer 2: ```suggestion
- Instale o [yarn](https://yarnpkg.com/pt-BR/) para obter as dependências e criar o projeto.
```

- Siga as instruções para a parte do projeto que você deseja alterar. (Veja seções específicas abaixo.)
- [Crie uma branch no Git](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) para isolar suas alterações:

  ```shell
  git checkout -b some-change
  ```

* Assim que tiver alterações no Git que você deseje enviar, [adicione-os e crie um commit](https://help.github.com/pt/articles/adding-a-file-to-a-repository-using-the-command-line). Para obter informações sobre como estruturar seus commits, consulte a documentação [Gerenciando PRs](/contributing/managing-pull-requests/#commit-and-pr-title).

>   - Usando um caractere de ponto `.` adicionará todos os arquivos não rastreados no diretório e subdiretórios atuais.

Reviewer 2: ```suggestion
  - O caractere ponto `.` adiciona todos os arquivos não rastreados no diretório e subdiretórios atuais.
```

  ```shell
  git add .
  ```

>   - Usando uma ferramenta visual como [GitHub Desktop](https://desktop.github.com/) ou [GitX](https://rowanj.github.io/gitx/) pode ajudar a escolher quais arquivos e linhas commitar.

Reviewer 2: ```suggestion
  - O uso de uma ferramenta visual como o [GitHub Desktop](https://desktop.github.com/) ou o [GitX](https://rowanj.github.io/gitx/) pode lhe ajudar a escolher quais arquivos e linhas commitar.
```

> * Commitando código executará o linter automatizado usando o [Prettier](https://prettier.io). Para executar o linter manualmente, execute um script npm no diretório base do projeto:

Reviewer 2: ```suggestion
* Ao commitar o código, será executado o linter automatizado usando o [Prettier](https://prettier.io). Para executar o linter manualmente, execute um script npm no diretório base do projeto:
```

  ```shell
  npm run format
  ```
* Crie um commit com as alterações de linting antes de enviar por push, [alterando o commit anterior](https://help.github.com/pt/articles/changing-a-commit-message) ou adicionando um novo commit. Para mais informações sobre linting e testes, visite a documentação [Gerenciando PRs](/contributing/managing-pull-requests/#automated-checks).
  ```shell
  git commit --amend
  ```
- Envie suas alterações para seu fork, assumindo que ele esteja configurado como [`origin`](https://www.git-tower.com/learn/git/glossary/origin):
  ```shell
  git push origin head
  ```

> - Para abrir uma PR com suas alterações no repositório Gatsby, você pode usar o [GitHub Pull Request UI](https://help.github.com/pt/articles/creating-a-pull-request). Como alternativa, você pode usar a linha de comando: recomendamos [hub](https://github.com/github/hub) para isso.

Reviewer: Alterar artigo "uma PR" => "um PR"

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

### Documentação de PRs

O site da documentação do Gatsby.js está principalmente nos diretórios [www](https://github.com/gatsbyjs/gatsby/tree/master/www) e [docs](https://github.com/gatsbyjs/gatsby/tree/master/docs) no GitHub, incluindo documentações e conteúdos de tutoriais. Também existem alguns [exemplos no repositório Gatsby](https://github.com/gatsbyjs/gatsby/tree/master/examples) mencionado na documentação.

> Passos adicionais para documentar PR

Reviewer: Adicionar o ":" no final
```suggestion
Passos adicionais para documentar PR:
```

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/f5af213ee6289a6e0645e9f7d7ef16f0b3cfe22d

> - Para alteraçÕes docs-only, considere usar `git checkout -b docs/some-change` ou `git checkout -b docs-some-change`, pois isso irá encurtar o circuito do processo de CI e executar apenas as tarefas de linting.

Reviewer 2: ```suggestion
- Para alterações que envolvam apenas documentação, considere usar `git checkout -b docs/some-change` ou `git checkout -b docs-some-change`, para encurtar o processo de CI e executar apenas as tarefas de linting.
```

Mais instruções podem ser encontradas na página de [contribuições de documentações](/contributing/docs-contributions/).

### Alterações de código

Instruções para fazer alterações no código-fonte do Gatsby, testes, componentes internos, APIs, pacotes e muito mais podem ser encontradas nas documentações de contribuições sobre [configurando seu ambiente de desenvolvimento local](/contributing/setting-up-your-local-dev-environment/). Também há detalhes adicionais na página [Contribuições de código](/contributing/code-contributions/).

> ### Starters ou Site Showcase

Reviewer 2: ```suggestion
### _Starters_ ou Amostras de Sites
```

Existem páginas específicas sobre como contribuir para várias partes do ecossistema de Gatsby:

> - [Submissões ao Showcase](/contributing/site-showcase-submissions/)

Reviewer 2: ```suggestion
- [Submissões de amostras de site](/contributing/site-showcase-submissions/)
```

- [Biblioteca Starter](/contributing/submit-to-starter-library/)

### Blog posts

> Para o blog do Gatsby, é necessário executar sua ideia de conteúdo pela equipe do Gatsby antes de enviá-la. Para obter mais informações, consulte a página de [contribuições de blog e site](/contributing/blog-and-website-contributions/), incluindo como propor uma idéia e configurar o blog para ser executado localmente.

Reviewer 2: ```suggestion
Para o blog do Gatsby, é necessário que sua ideia de conteúdo seja aprovada pela equipe do Gatsby antes de ser enviada. Para obter mais informações, consulte a página de [contribuições de blog e site](/contributing/blog-and-website-contributions/), incluindo como propor uma ideia e configurar o blog para ser executado localmente.
```

## Atualize seu fork com as alterações mais recentes do Gatsby

> O repositório do Gatsby GitHub é muito ativo, portanto, é provável que você precise atualizar seu fork com as alterações mais recentes para poder mergir no seu código. Isso requer adicionar o Gatsby como um [upstream remote](https://help.github.com/pt/articles/configuring-a-remote-for-a-fork):

Reviewer 2: ```suggestion
O repositório do Gatsby GitHub é muito ativo, portanto, é provável que você precise atualizar seu fork com as alterações mais recentes para poder submeter no seu código. Isso requer adicionar o Gatsby como um [upstream remote](https://help.github.com/pt/articles/configuring-a-remote-for-a-fork):
```

> - Defina a URL do repositório do Gatsby como uma fonte remota. O nome do remoto é arbitrári; este exemplo usa `upstream`.

Reviewer: "arbitrári" => "arbitrária"

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/f5af213ee6289a6e0645e9f7d7ef16f0b3cfe22d

  ```
- [Na branch](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging) que você deseja atualizar, faça merge de todas as alterações do Gatsby no seu fork:
  ```shell
  git merge upstream/master
  ```
  - Se houver alguns [conflitos de mesclagem](https://help.github.com/pt/articles/resolving-a-merge-conflict-on-github), você vai querer resolver-los para obter um merge limpo.
- Quando sua branch estiver funcionando em boas condições, faça push das alterações para o seu fork:
  ```shell
  git push origin head
  ```

Para mais informações sobre como trabalhar com repositórios upstream, [visite a documentação do GitHub](https://help.github.com/pt/articles/configuring-a-remote-for-a-fork).

_**Nota:** como membro do repositório Gatsby, você também pode cloná-lo diretamente (em vez de fazer um fork e usar um fluxo de trabalho remoto upstream). Você pode então enviar via push as alterações para [feature branches](https://git-scm.com/book/en/v1/Git-Branching-Branching-Workflows) para abrir PRs._

## Recursos adicionais

- CSS Tricks: [How to Contribute to an Open Source Project](https://css-tricks.com/how-to-contribute-to-an-open-source-project/)

> - [Criar uma pull request](https://help.github.com/pt/articles/creating-a-pull-request) do GitHub

Reviewer: Alterar artigo "uma" => "um"

Author: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/3b3c21095fb45f6be256aa235cc1b8677bc185fa

> - [Configurar remote para bifurcação](https://help.github.com/pt/articles/configuring-a-remote-for-a-fork)

Reviewer: "bifurcação" => "fork"

Você está utilizando fork em todo o restante do doc.

Author: Por ser uma referência, eu trouxe o título do artigo:
![image](https://user-images.githubusercontent.com/5200384/68024272-657db600-fc88-11e9-9d5f-471826802dd6.png)
https://help.github.com/pt/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork

Author: Done: https://github.com/gatsbyjs/gatsby-pt-BR/pull/29/commits/af1ac2d31b6cca5b70b0a1beeaba1b9ae79b8735

Reviewer: Ah beleza! sem problemas então.
